nodeadm/internal/kubelet/config.go

```diff
@@ -268,6 +267,21 @@ func (k *kubelet) GenerateKubeletConfig(cfg *api.NodeConfig) (*kubeletSubConfig,
 		kubeletConfig.withDefaultReservedResources()
 	}
 
+	if len(cfg.Spec.Kubelet.Labels) > 0 {
+		var labelStrings []string
+		for labelKey, label := range cfg.Spec.Kubelet.Labels {
+			labelStrings = append(labelStrings, fmt.Sprintf("%s=%s", labelKey, label))
+		}
+		k.additionalArguments["node-labels"] = strings.Join(labelStrings, ",")
```

Comment:
Would like to double confirm these keys are supported for all versions?
`node-labels` and `register-with-taints`

Reply:
checked [changelogs](https://github.com/kubernetes/kubernetes/tree/master/CHANGELOG) and only found a couple of relevant details:
* [1.23 : `--register-with-taints` option is now available via the Kubelet config file field registerWithTaints](https://github.com/kubernetes/kubernetes/blob/7bb00356f06332e63a9f06acd42f1bdd8fc559d6/CHANGELOG/CHANGELOG-1.23.md?plain=1#L2401)
  * seems safe to use for now in the case that a cluster beyond 1.23 is spun up
* no important changes for `--node-labels`

Reply:
however, does ^ incentivize us to put this in `Kubelet.Config.RegisterWithTaints` instead to keep on the edge of kubelet config adoption?

Reply:
Here's what we do in Karpenter, which has worked out very well: https://github.com/kubernetes-sigs/karpenter/blob/main/pkg/apis/v1beta1/nodeclaim.go#L70

Reply:
think this is a good idea, and will treat the config as first class. i.e. write the `--register-with-taints` argument if necessary based on a version check. similar to sichao's [container runtime catch](https://github.com/awslabs/amazon-eks-ami/blob/al2023/nodeadm/internal/kubelet/config.go#L209)